cmake/zephyr_module.cmake

```diff
@@ -0,0 +1,83 @@
+# This cmake file provides functionality to import additional out-of-tree, OoT
+# CMakeLists.txt and Kconfig files into Zephyr build system.
+# It uses -DZEPHYR_MODULES=<oot-path-to-module>[;<additional-oot-module(s)>]
+# given to CMake for a list of folders to search.
+# It looks for: <oot-module>/zephyr/module.yml or
+#               <oot-module>/zephyr/CMakeLists.txt
+# to load the oot-module into Zephyr build system.
+# If west is available, it uses `west list` to obtain a list of projects to
+# search for zephyr/module.yml
+
+# Clear zephyr modules property
+set_property(GLOBAL PROPERTY ZEPHYR_MODULES_NAME "")
+set_property(GLOBAL PROPERTY ZEPHYR_MODULES_DIR  "")
+
+if(ZEPHYR_MODULES)
+  set(west_project_list ${ZEPHYR_MODULES})
+else()
+  ## Use `west list` to fetch all west handled projects.
+  execute_process(
+    COMMAND
+    west list --format={abspath}\;
+    OUTPUT_VARIABLE
+    west_project_list
+  )
+  string(REGEX REPLACE "[\r\n]+" ";" west_project_list "${west_project_list}")
+endif()
+
+if(ZEPHYR_EXTRA_MODULES)
+  list(APPEND west_project_list ${ZEPHYR_EXTRA_MODULES})
+endif()
+
+# Clear the Kconfig.modules generated file in case modules has been removed.
+# The Kconfig.modules contains a list of additional Kconfig files to be sourced
+# based upon <module>/zephyr/module.yml files.
+file(WRITE ${PROJECT_BINARY_DIR}/Kconfig.modules
```

Comment:
What is `PROJECT_BINARY_DIR`? is it `build/zephyr`?

Reply:
Yes.

Reply:
And that is already used else-where in Kconfig.
So I decided to use same property in this PR, instead of exporting yet another property.